pkg/build/builder/cmd/builder.go

```diff
@@ -51,16 +48,17 @@ func newBuilderConfigFromEnvironment(out io.Writer, needsDocker bool) (*builderC
 	cfg.out = out
 
 	buildStr := os.Getenv("BUILD")
-	cfg.build = &buildapi.Build{}
 
-	obj, groupVersionKind, err := legacyscheme.Codecs.UniversalDecoder().Decode([]byte(buildStr), nil, nil)
+	cfg.build = &buildapiv1.Build{}
+
+	obj, groupVersionKind, err := legacyscheme.Codecs.UniversalDecoder().Decode([]byte(buildStr), nil, cfg.build)
```

Comment:
The legacy scheme keeps you pinned to the internal versions to operate.  You'll need a follow up to remove the dependency.  Definitely not a blocker.

Reply:
is there a different decoder/scheme I should be using instead?